src/tezos_interop/tezos_interop.ml

```diff
@@ -179,18 +143,18 @@ module Listen_transactions = struct
           on_fail in
       read_line_until_fails ()
   end
-  let listen ~context ~destination ~on_message =
+  let listen t ~destination ~on_message =
     let rec start () =
       Lwt.catch
-        (fun () -> CLI.run ~context ~destination ~on_message ~on_fail)
+        (fun () -> CLI.run t ~destination ~on_message ~on_fail)
         (fun _exn -> on_fail ())
     and on_fail () = start () in
     Lwt.async start
 end
 module Consensus = struct
   open Michelson.Michelson_v1_primitives
   open Tezos_micheline
-  let commit_state_hash ~context ~block_height ~block_payload_hash ~state_hash
+  let commit_state_hash t ~block_height ~block_payload_hash ~state_hash
       ~withdrawal_handles_hash ~validators ~signatures =
     let module Payload = struct
       type t = {
```

Comment:
I am confused about this type being defined within `commit_state_hash` and not "globally", would you mind explain me why this is the good thing to do here instead of splitting `Payload` to a dedicated module?

Reply:
For what purpose? The structure of the payload is only needed at the interface between the smart contract and Deku, so I think it makes sense to keep it relatively private.

Reply:
That's exactly it, it would be reasonable to move it to `Consensus.Payload` but by default just put stuff where you're gonna use it.